File: 1/fuel.cpp
#include <iostream>
#include <fstream>
using namespace std;

int calc(int num) {
    return num / 3 - 2;
}

int calc_with_fuel(int num) {
    int tmp = calc(num);

    if(tmp <= 0) {
        return 0;
    }

    return tmp + calc_with_fuel(tmp);
}

int main () {
    ifstream myfile;
    myfile.open("in.txt");

    int tmp, sum = 0, sum_rek = 0;

    for(int i = 0; i < 100; i++) {
        myfile >> tmp;
        sum += calc(tmp);
        sum_rek += calc_with_fuel(tmp);
    }

    myfile.close();

    cout << sum << endl << sum_rek << endl;

    return 0;
}

File: 1/fuel.js
const fs = require('fs');

const calc = (data) => Math.floor(parseInt(data) / 3) - 2;

const solve_first = (lines) => {
    const out = lines.reduce((prev, curr) => prev + calc(curr), 0);
    console.log(out);
}

const calc_with_fuel = (number) => {
    const tmp = Math.floor(parseInt(number) / 3) - 2;

    if (tmp <= 0) {
        return 0;
    }

    return tmp + calc_with_fuel(tmp);
}

const solve_second = (lines) => {
    const data = lines.map(calc_with_fuel);
    const out = data.reduce((prev, curr) => prev + curr, 0);
    console.log(out);
}

fs.readFile('in.txt', { encoding: 'utf-8' }, (err, data) => {
    if (err) {
        console.error(err);
        return;
    }

    const lines = data.split('\n');

    solve_first(lines);
    solve_second(lines);
});
